src/main/java/de/epiceric/shopchest/listeners/ShopInteractListener.java

```diff
@@ -691,18 +691,8 @@ private void create(final Player executor, final Location location, final ItemSt
             executor.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.SHOP_CREATED, placeholder));
         }
 
-        for (Player p : location.getWorld().getPlayers()) {
-            if (p.getLocation().distanceSquared(location) <= Math.pow(config.maximal_distance, 2)) {
-                if (shop.getHologram() != null) {
-                    shop.getHologram().showPlayer(p);
-                }
-            }
-            if (p.getLocation().distanceSquared(location) <= Math.pow(config.maximal_item_distance, 2)) {
-                if (shop.getItem() != null) {
-                    shop.getItem().setVisible(p, true);
-                }
-            }
-        }
+        // next update will display the new shop
+        plugin.getShopUtils().resetPlayerLocation(executor);
```

Comment:
Then every time someone in the same world as you creates a shop, the items for every shop will be respawned, which might look odd for someone looking on a shop. It would be better if only the newly created shop will be updated.